File: view of  Binary tree/right view.cpp
// using BFS
// Time: O(n) 
// space: q size will be at max n /2 so space complexity is O(n)

// Using preorder recursion
// but here we will treverse right right, then left
// time: O(n) and space: O(H of Tree)
class Solution {
public:
    vector<int> solve(TreeNode*root, int level, vector<int> &result){
        if (!root){
            return result;
        }
        if (level == result.size()){
            result.push_back(root->val);
        }
        solve(root->right, level + 1, result);
        solve(root->left, level + 1, result);
        return result;
    }
    vector<int> rightSideView(TreeNode* root) {
        vector<int> result;
        return solve(root, 0, result);
    }
};

File: view of  Binary tree/right view.py
# using BFS
# Time: O(n) 
# space: q size will be at max n /2 so space complexity is O(n)

# with preorder recursion
# but here we treverse right subtree first, then right
#  time: O(n) and space: O(H of binary Tree)
class Solution:
    
    def solve(self, root, level, result):
        if root is None:
            return result
        if len(result) == level:
            result.append(root.val)
        self.solve(root.right, level + 1, result)
        self.solve(root.left, level + 1, result)
        return result
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        return self.solve(root, 0, [])
        
